//
// This file is roughly based on
// the extract section of
// https://github.com/RTimothyEdwards/open_pdks/blob/master/sky130/magic/sky130.tech
//
syntax = "proto3";

package kpex.tech;

message ProcessParasiticsInfo {
    // we do all in µm
    
    int32 side_halo = 10;
    
    ResistanceInfo resistance = 110;
    CapacitanceInfo capacitance = 111;
}


message ResistanceInfo {
    
    // resistances are in milliohm / µm^2
    message LayerResistance {
        string layer_name = 1;
        double resistance = 2;
        double corner_adjustment_fraction = 3;
    }
    
    // resistances are in milliohm / µm^2
    message ViaResistance {
        string via_name = 1;
        double resistance = 2;
    }
    
    repeated LayerResistance layers = 1;
    repeated ViaResistance vias = 2;
}

message CapacitanceInfo {

    message SubstrateCapacitance {
        string layer_name = 1;
        double area_capacitance = 2;       // in attoFarad / µm^2
        double perimeter_capacitance = 3;  // in attoFarad / µm
    }

    message OverlapCapacitance {
        string top_layer_name = 1;
        string bottom_layer_name = 2;
        double capacitance = 3;  // in attoFarad / µm
    }

    message SidewallCapacitance {
        string layer_name = 1;
        double capacitance = 2;  // in attoFarad / µm
        double offset = 3;
    }

    message SideOverlapCapacitance {
        string in_layer_name = 1;
        string out_layer_name = 2;
        double capacitance = 3;  // in attoFarad / µm
    }

    repeated SubstrateCapacitance substrates = 200;
    repeated OverlapCapacitance overlaps = 201;
    repeated SidewallCapacitance sidewalls = 202;
    repeated SideOverlapCapacitance sideoverlaps = 203;
}

// ----------------------------------------------------------------------------------


message StyleVariant {
    
}

